formatter/src/main/java/org/togetherjava/formatter/Formatter.java

```diff
@@ -0,0 +1,319 @@
+                if (type == TokenType.OPEN_BRACES) {
+                    if (lastToken == TokenType.CLOSE_PARENTHESIS
+                            || lastToken == TokenType.IDENTIFIER || lastToken == TokenType.TRY || lastToken == TokenType.ELSE) {
+                        result.append(" ");
+                    }
+
+                    append(result, token);
+
+                    result.append("\n");
+
+                    lastNewLine = true;
+                    indentation++;
+
+                    break iteration;
+                } else if (lastToken == TokenType.CLOSE_BRACKETS && type != TokenType.SEMICOLON) {
+                    result.append(" ");
+
+                    append(result, token);
+
+                    break iteration;
+                } else if (type == TokenType.CLOSE_BRACES) {
+                    result.append("\n");
+
+                    putIndentation(--indentation, result);
+
+                    append(result, token);
+
+                    result.append("\n");
+
+                    lastNewLine = true;
+
+                    break iteration;
+                } else if (type == TokenType.IDENTIFIER && (lastToken == TokenType.CLOSE_BRACES
+                        || lastToken == TokenType.OPEN_BRACES
+                        || lastToken == TokenType.SEMICOLON
+                        || lastToken == TokenType.UNKNOWN)) {
+                    append(result, token);
+
+                    break iteration;
+                } else if (type == TokenType.COMMENT || type == TokenType.SEMICOLON
+                        || type == TokenType.ANNOTATION) {
+                    if (type == TokenType.COMMENT) {
+                        result.append(" ");
+                    }
+
+                    append(result, token);
+
+                    if (forClosed) {
+                        result.append("\n");
+
+                        lastNewLine = true;
+                    }
+
+                    break iteration;
+                } else if ((type == TokenType.OPEN_PARENTHESIS && lastToken != TokenType.IDENTIFIER
+                        && lastToken != TokenType.NOT && lastToken != TokenType.BIGGER
+                        && lastToken != TokenType.OPEN_PARENTHESIS) || lastToken == TokenType.PUBLIC
+                        || lastToken == TokenType.PRIVATE || lastToken == TokenType.PROTECTED
+                        || (lastToken == TokenType.BIGGER && type != TokenType.OPEN_PARENTHESIS)
+                        || lastToken == TokenType.RETURN || lastToken == TokenType.COMMA
+                        || lastToken == TokenType.CATCH
+                        || (lastToken == TokenType.CLOSE_PARENTHESIS
+                                && type != TokenType.CLOSE_PARENTHESIS && type != TokenType.DOT
+                                && type != TokenType.COMMA)
+                        || lastToken == TokenType.PACKAGE) {
+                    result.append(" ");
+                }
+
+                if (!forClosed) {
```

Comment:
I'd rather call a different format method like `formatFor()` in this case, that cuts down on the amount of variables and makes it easier to read.

Keep in mind that you can make the whole class stateful, i.e. have a token queue as a *field* you walk through - then you do not need to have as many variables and can decompose the formatting into (possibly recursive) functions where needed. Or you just have helper functions like `formatFor`, that parse a few tokens and then yield control to the caller again.